plugins/woocommerce-blocks/assets/js/blocks/product-collection/edit/inspector-controls/related-by-control.tsx

```diff
@@ -0,0 +1,67 @@
+/**
+ * External dependencies
+ */
+import { __ } from '@wordpress/i18n';
+import { CheckboxControl, PanelBody } from '@wordpress/components';
+import { recordEvent } from '@woocommerce/tracks';
+
+/**
+ * Internal dependencies
+ */
+import { QueryControlProps, RelatedBy, CoreFilterNames } from '../../types';
+
+const RelatedByControl = ( {
+	query,
+	setQueryAttribute,
+	trackInteraction,
+}: QueryControlProps ) => {
+	const relatedBy = query?.relatedBy as RelatedBy;
+
+	const handleRelatedByChange = (
+		value: boolean,
+		type: 'categories' | 'tags'
+	) => {
+		const newRelatedBy = {
+			...relatedBy,
+			[ type ]: value,
+		};
+
+		setQueryAttribute( {
+			relatedBy: newRelatedBy,
+		} );
+
+		trackInteraction( CoreFilterNames.RELATED_BY );
+
+		recordEvent(
+			'blocks_product_collection_inspector_control_related_by_changed',
+			{
+				categories: newRelatedBy?.categories || false,
+				tags: newRelatedBy?.tags || false,
+			}
+		);
```

Comment:
Do we want a new event just for this? 🤔 If needed maybe we can collect more data within the interaction above although the assumption was we don't need information what's the value merchants choose but the number of interaction matters.

Reply:
While I don't have a strong opinion on this, I see potential value in keeping the tracking data:

1. It could help us understand how merchants are using the related products feature, particularly which relationship types (categories vs tags) they prefer

2. Even if we don't have immediate plans to analyze this data, having it available could be useful for future decision-making around feature improvements

3. Since we already have the tracking infrastructure in place, the overhead of collecting this additional data point is minimal

That said, I'm open to removing it if others feel it's unnecessary. I mainly added it based on @octaedro's [suggestion](https://github.com/woocommerce/woocommerce/pull/52580#discussion_r1831126066).

Reply:
Personally I'm not convinced 😅 I had discussion about the tracking with @octaedro and we clearly have different approach 😅 

> It could help us understand how merchants are using the related products feature, particularly which relationship types (categories vs tags) they prefer

I can't think of single usage atm. Because let's imaging scenario user plays with the setting:
- clicks categories 8 times
- clicks tags 2 times
They end up with default setup. We cannot conclude from those numbers categories are more popular.

To have a meaningful data what merchants prefer, we should actually extend logic by @xristos3490 from this PR: https://github.com/woocommerce/woocommerce/pull/46466.

> Even if we don't have immediate plans to analyze this data, having it available could be useful for future decision-making around feature improvements

I see, can't argue with that. Although I prefer more purposeful approach where we say "We want to know X, let's measure Y". In here, I cannot see potential use cases is all 😅 

> Since we already have the tracking infrastructure in place, the overhead of collecting this additional data point is minimal

True implementation-wise, but we need to remember collecting data long-term comes with the cost (storing etc.)

---

Overall I won't be fighting for removal of this however:
- I'd love to understand one potential use case for it
- I believe extending https://github.com/woocommerce/woocommerce/pull/46466 instead would give us more value.

Reply:
Hi @imanish003 and @kmanijak, this is an interesting thread.

Here are my two cents. Collecting data for new functionalities is usually a good idea. In this case, depending on how users interact with the checkboxes, we can prioritize or deprioritize the work done there.

If nobody interacts with the checkboxes, we'll know that improving them is not a priority. Conversely, if we see a high level of interaction, it may indicate that the component is not as intuitive as we thought. In such cases, adding a text explanation on how it works could be beneficial.

> True implementation-wise, but we need to remember collecting data long-term comes with the cost (storing etc.)

While it is true that any data collection process involves costs associated with storage, modern technologies and efficient data management practices make these costs manageable and often offset by the value that the data provided.

Having said that, if you think that collecting this data is not useful, I'm okay with not recording it or only recording the interactions, regardless of the selected items.

I would like to know @j111q's opinion on this.

Reply:
Don't get me wrong! I'm ok with collecting data but I'm suggesting different approach (tracking outcome) which IMO gives more information and is in sync with how we track all the other filters/controls.

Reply:
Great to see such a productive discussion. Everyone is raising their concerns and providing thoughtful perspectives on the tracking approach.  Thank you 🙌🏻

> To have a meaningful data what merchants prefer, we should actually extend logic by @xristos3490 from this PR: #46466.

I agree that the current tracking approach doesn't provide meaningful insights into merchant behavior. The saved filter values in post would be much more valuable for understanding how the feature is actually being used.

After reviewing the PR by @xristos3490, I notice it only tracks which features/filters are enabled, without capturing their configured values. While extending the tracking to include these values would be valuable, I believe this enhancement deserves its own dedicated task rather than being added to this PR. Because we should maintain consistency by tracking filter values across all Product Collection controls, not just this one. Also, It would significantly expand this PR's scope beyond its core purpose.

I suggest we create a separate task to implement comprehensive filter value tracking. What are your thoughts on this approach? CC: @kmanijak @octaedro

> If nobody interacts with the checkboxes, we'll know that improving them is not a priority.

@octaedro We actually already have tracking in place for this via the `blocks_product_collection_inspector_control_clicked` event [here](https://github.com/woocommerce/woocommerce/blob/3cd67d1e019116f77b84cfb662a45e3c68f0eacb/plugins/woocommerce-blocks/assets/js/blocks/product-collection/edit/inspector-controls/related-by-control.tsx#L33). This existing tracking should give us good visibility into whether users are interacting with these controls or not. Isn't it?

Reply:
Based on above discussion, I have created a new issue #52696 to implement control values tracking.